expression/builtin_time_test.go

```diff
@@ -2761,3 +2761,29 @@ func (s *testEvaluatorSuite) TestTidbParseTso(c *C) {
 		c.Assert(d.IsNull(), IsTrue)
 	}
 }
+
+func (s *testEvaluatorSuite) TestGetIntervalFromDecimal(c *C) {
+	defer testleak.AfterTest(c)()
+	du := baseDateArithmitical{}
+
+	tests := []struct {
+		param  string
+		expect string
+		unit   string
+	}{
+		{"1.100", "1:100", "MINUTE_SECOND"},
+		{"1.10000", "1-10000", "YEAR_MONTH"},
+		{"1.10000", "1 10000", "DAY_HOUR"},
+		{"11000", "0 00:00:11000", "DAY_MICROSECOND"},
+		{"11000", "00:00:11000", "HOUR_MICROSECOND"},
+		{"11.1000", "00:11:1000", "HOUR_SECOND"},
+		{"1000", "00:1000", "MINUTE_MICROSECOND"},
+	}
+
+	for _, test := range tests {
+		interval, isNull, err := du.getIntervalFromDecimal(s.ctx, s.datumsToConstants([]types.Datum{types.NewDatum("CURRENT DATE"), types.NewDecimalDatum(newMyDecimal(c, test.param))}), chunk.Row{}, test.unit)
```

Comment:
https://github.com/pingcap/tidb/commit/ae58fd01f11117c6a8ad3489c825665afa79ccce has been cherry-picked in https://github.com/pingcap/tidb/pull/11560, PTAL